source/boundary_traction/interface.cc

```diff
@@ -73,59 +86,232 @@ namespace aspect
+        traction += plugin->boundary_traction(boundary_indicator,
+                                              position,normal_vector);
+
+      return traction;
     }
 
 
 
     template <int dim>
-    std::string
-    get_names ()
+    const std::map<types::boundary_id, std::pair<std::string,std::vector<std::string>>> &
+    Manager<dim>::get_active_boundary_traction_names () const
     {
-      return std::get<dim>(registered_plugins).get_pattern_of_names ();
+      return boundary_traction_indicators;
+    }
+
+    template <int dim>
+    const std::map<types::boundary_id,std::vector<std::unique_ptr<BoundaryTraction::Interface<dim>>>> &
+    Manager<dim>::get_active_boundary_traction_conditions () const
+    {
+      return boundary_traction_objects;
     }
 
 
 
     template <int dim>
     void
-    declare_parameters (ParameterHandler &prm)
+    Manager<dim>::declare_parameters (ParameterHandler &prm)
     {
+      prm.enter_subsection ("Boundary traction model");
+      {
+        prm.declare_entry ("Prescribed traction boundary indicators", "",
+                           Patterns::Map (Patterns::Anything(),
+                                          Patterns::Selection(std::get<dim>(registered_plugins).get_pattern_of_names ())),
+                           "A comma separated list denoting those boundaries "
+                           "on which the traction is prescribed, i.e., where unknown "
+                           "external forces act to prescribe a particular traction. This is "
+                           "often used to prescribe a traction that equals that of "
+                           "overlying plates."
+                           "\n\n"
+                           "The format of valid entries for this parameter is that of a map "
+                           "given as ``key1 [selector]: value1, key2 [selector]: value2, key3: value3, ...'' where "
+                           "each key must be a valid boundary indicator (which is either an "
+                           "integer or the symbolic name the geometry model in use may have "
+                           "provided for this part of the boundary) "
+                           "and each value must be one of the currently implemented boundary "
+                           "traction models. ``selector'' is an optional string given as a subset "
+                           "of the letters `xyz' that allows you to apply the boundary conditions "
+                           "only to the components listed. As an example, '1 y: function' applies "
+                           "the type `function' to the y component on boundary 1. Without a selector "
+                           "it will affect all components of the traction."
+                           "\n\n"
+                           "Note that when ``Use years in output instead of seconds'' is set "
+                           "to true, traction should be given in N/m^2. "
```

Comment:
Time has not influence on the unit of tractions.
```suggestion
                           "Note that traction should be given in N/m^2. "
```